src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -77,12 +79,21 @@ public BusinessesService(List<Listing> allBusinesses) {
             .apiKey(KEY)
             .build();
     TextSearchRequest request = PlacesApi.textSearchQuery(context, product);
-    
+    int radius = 1500;
+
     try {
       PlacesSearchResponse response = request.location(latLng)
-              .radius(TEXT_SEARCH_RADIUS)
+              .radius(radius)
               .await();
-      
+
+      while (response.results.length < MAX_NUMBER_OF_RESULTS_PER_REQUEST && 
+             radius < MAX_ALLOWED_TEXT_SEARCH_RADIUS) {
+        // If the list of businesses are not enough, the radius is increased
```

Comment:
nit: the wording of this comment is a bit awkward - I'm not sure the comment is necessary. I think it could be something as simple as "Increase radius if there are less than 20 results"

Reply:
Thank you for this and I have changed the comment